### PR TITLE
Update old VS refs

### PR DESCRIPTION
**Pull request by gewarren**
- remove refs to Visual Studio 2005
- remove leading/trailing whitespace (if this creates too much noise, let me know - hopefully it diffs better in CodeFlow)

Contributes to  #6923 and #7172

### CODE REVIEW

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this looks great @gewarren  (And, reviewing it sent me to the C# spec twice. 🤔 )

I had a few small comments that we should address before merging.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @gewarren I've approved the changes.

File: docs/csharp/language-reference/keywords/foreach-in.md [hunk 1]
```diff
@@ -41,10 +41,10 @@ Beginning with C# 7.3, if the enumerator's `Current` property returns a [referen
...
-[C# Keywords](index.md)  
-[C# Reference](../index.md)  
-[C# Programming Guide](../../programming-guide/index.md)  
+[The foreach statement (C# language specification)](/dotnet/csharp/language-reference/language-specification/statements#the-foreach-statement)
+[Using foreach with Arrays](../../programming-guide/arrays/using-foreach-with-arrays.md)
+[for](for.md)
+[Iteration Statements](iteration-statements.md)
+[C# Keywords](index.md)
+[C# Reference](../index.md)
+[C# Programming Guide](../../programming-guide/index.md)
```
Comment (pkulikov) [inline docs/csharp/language-reference/keywords/foreach-in.md]: Same comment as above: either to bring back the trailing white spaces or format the **See also** links as a list.

Comment (mairaw) [reply docs/csharp/language-reference/keywords/foreach-in.md]: We recently standardized on formatting see also as list. See #6923

File: docs/csharp/language-reference/keywords/switch.md [hunk 2]
```diff
@@ -159,35 +162,33 @@ Note that `null` does not match a type. To match a `null`, you use the following
...
 [!code-csharp[switch#8](../../../../samples/snippets/csharp/language-reference/keywords/switch/when-clause.cs#1)]
-  
+
 Note that the `when` clause in the example that attempts to test whether a `Shape` object is `null` does not execute. The correct type pattern to test for a `null` is `case null:`.
 
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](../../../../includes/csharplangspec-md.md)]  
-  
-## See Also  
+## C# Language Specification
```
Comment (mairaw) [inline docs/csharp/language-reference/keywords/switch.md]: wanna format this and the next heading as sentence case as well?

File: docs/csharp/programming-guide/concepts/iterators.md [hunk 3]
```diff
@@ -4,365 +4,355 @@ ms.date: 07/20/2015
...
-## See Also  
- <xref:System.Collections.Generic>  
- <xref:System.Collections.Generic.IEnumerable%601>  
- [foreach, in](../../../csharp/language-reference/keywords/foreach-in.md)  
- [yield](../../../csharp/language-reference/keywords/yield.md)  
- [Using foreach with Arrays](../../../csharp/programming-guide/arrays/using-foreach-with-arrays.md)  
- [Generics](../../../csharp/programming-guide/generics/index.md)
+> For all examples in this topic except the Simple Iterator example, include [using](../../../csharp/language-reference/keywords/using-directive.md) directives for the `System.Collections` and `System.Collections.Generic` namespaces.
+
+##  <a name="BKMK_SimpleIterator"></a> Simple Iterator
```
Comment (mairaw) [inline docs/csharp/programming-guide/concepts/iterators.md]: I'd remove the anchors too

File: docs/csharp/language-reference/keywords/foreach-in.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/keywords/foreach-in.md]: @mairaw thank you for the update; then the list formatting should be added in both documents.

**Comment by mairaw** [comment]
Don't forget to change ms.date as you're reviewing these topics as well.

File: docs/csharp/language-reference/keywords/for.md [hunk 4]
```diff
@@ -90,23 +90,23 @@ i++
...
-[C# Programming Guide](../../programming-guide/index.md)  
-[C# Keywords](index.md)  
-[foreach, in](foreach-in.md)  
-[for Statement (C++)](/cpp/cpp/for-statement-cpp)  
+[The for statement (C# language specification)](/dotnet/csharp/language-reference/language-specification/statements#the-for-statement)
+[C# Reference](../index.md)
+[C# Programming Guide](../../programming-guide/index.md)
+[C# Keywords](index.md)
+[foreach, in](foreach-in.md)
+[for Statement (C++)](/cpp/cpp/for-statement-cpp)
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/csharp/language-reference/keywords/for.md]: Oops, thanks for catching that. I have my VSCode set to remove trailing whitespace so it happens automatically.

[the remark this answers is not in the document]
Comment (mairaw) [reply docs/csharp/language-reference/keywords/for.md]: That's why we're changing the guidance 😄

File: docs/csharp/language-reference/keywords/for.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "for (C# Reference)"
+title: C# for statement
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/for.md]: I changed a few titles to improve SEO. Let me know if you don't want me to change these though.

Comment (BillWagner) [reply docs/csharp/language-reference/keywords/for.md]: I like this.  I agree it will be more findable.

**Comment by gewarren** [comment]
#sign-off

**Pull request merged**